extensions/amp-date-display/1.0/amp-date-display.js

```diff
@@ -35,6 +36,16 @@ class AmpDateDisplay extends BaseElement {
     this.template_ = null;
   }
 
+  /** @override */
+  buildCallback() {
+    super.buildCallback();
+    // Handle additional JSON
+    const scriptElement = childElementsByTag(this.element, 'SCRIPT')[0];
+    if (scriptElement && isJsonScriptTag(scriptElement)) {
+      this.element.setAttribute('json', scriptElement.textContent);
+    }
```

Comment:
Is there a reason this logic is in `amp-date-display.js` (AMP-only layer) over `base-element.js` (generic layer)?